src/btree/bt_compact.c

```diff
@@ -368,7 +368,9 @@ __wt_compact(WT_SESSION_IMPL *session)
          * Compact pulls pages into cache during the walk without checking whether the cache is
          * full. Check now to throttle compact to match eviction speed.
          */
-        WT_ERR(__wt_cache_eviction_check(session, false, false, NULL));
+        WT_ERR(__wt_cache_eviction_check(session, false, false, &didwork));
```

Comment:
At what point are we able to exit from this function? My concern is the common question of "compact is running for a long time and I haven't reclaimed any space".

Reply:
I think we're back to the discussion from this commit https://github.com/wiredtiger/wiredtiger/pull/3707 -- Does compaction a bad idea if there's eviction pressure? 

Since if cache is under pressure, compaction can only do `run for a long time` or `WT_ERR(EBUSY)`, if my understanding is correct there are no other better options. What do you think @etienneptl should I pull someone else to be in this discussion?

Reply:
> At what point are we able to exit from this function?

We need to look into the function but like any thread that is evicting, it can spend enough time to make compact slow.

I think, by default, we should let any application thread be pulled into eviction - WiredTiger has been following this design. If the application is against it, it is possible to create a session with the flag that sets `WT_SESSION_IGNORE_CACHE_SIZE`. This makes me wonder, if we want to keep the same behaviour for the server, should make them create a session with such a flag?
@clarissecheah and @GWlodarek can help here.

@keitharnoldsmith, what are your thoughts on this?

Reply:
I see we count the number of pages we evict [here](https://github.com/wiredtiger/wiredtiger/blob/d67b12877ac155ff8797cb507febb866c68848ee/src/evict/evict_lru.c#L2707-L2718) 
```
        /*
         * Check if we have become busy.
         *
         * If we're busy (because of the transaction check we just did or because our caller is
         * waiting on a longer-than-usual event such as a page read), and the cache level drops
         * below 100%, limit the work to 5 evictions and return. If that's not the case, we can do
         * more.
         */
        if (!busy && __wt_atomic_loadv64(&txn_shared->pinned_id) != WT_TXN_NONE &&
          __wt_atomic_loadv64(&txn_global->current) != __wt_atomic_loadv64(&txn_global->oldest_id))
            busy = true;
        max_progress = busy ? 5 : 20;
```
limiting the number of pages we evict to 5 or 20, depending on whether we're busy. Although, I'm not certain what's meant by busy here. I don't think this will greatly affect the compact walk since this check is only done when we read in internal pages. So I'm happy to leave as is!

Reply:
As discussed I will let Greg know what we've changed for compaction, and there could be customers' option to set session's WT_SESSION_IGNORE_CACHE_SIZE so whoever calls the foreground compaction can be aware of it.